tests/trace_state_propagation.rs

```diff
@@ -158,15 +158,14 @@ fn build_sampled_context() -> (Context, impl Subscriber, TestExporter, TracerPro
 #[derive(Clone, Default, Debug)]
 struct TestExporter(Arc<Mutex<Vec<SpanData>>>);
 
-#[async_trait]
 impl SpanExporter for TestExporter {
-    async fn export(
-        &mut self,
-        mut batch: Vec<SpanData>,
-    ) -> opentelemetry::sdk::export::trace::ExportResult {
-        if let Ok(mut inner) = self.0.lock() {
-            inner.append(&mut batch);
-        }
-        Ok(())
+    fn export(&mut self, mut batch: Vec<SpanData>) -> BoxFuture<'static, ExportResult> {
```

Comment:
This seems like a drive-by change, is there some reason behind it other than personal preference? Does it allow removing async-trait from dependencies?

Reply:
@tomkarw this is from #9 which is an exact backport from the main repo, I am not the original author of it

Reply:
Got it. It's a nitpick, so I don't think chasing the original author makes sense.